src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,245 @@
+"""DSM external USB device data."""
+
+from __future__ import annotations
+
+from synology_dsm.api import SynoBaseApi
+from synology_dsm.helpers import SynoFormatHelper
+
+
+class SynoCoreExternalUSB(SynoBaseApi):
+    """Class for external USB storage devices."""
+
+    API_KEY = "SYNO.Core.ExternalDevice.Storage.USB"
+    REQUEST_DATA = {"additional": '["all"]'}
+
+    async def update(self):
+        """Updates external USB storage device data."""
+        raw_data = await self._dsm.post(self.API_KEY, "list", data=self.REQUEST_DATA)
+        if raw_data:
+            device_data = raw_data["data"]
+            for device in device_data["devices"]:
+                self._data[device["dev_id"]] = SynoCoreExternalUSBDevice(device)
+
+    # Root
+    @property
+    def get_devices(self):
+        """Gets all external USB storage devices."""
+        return self._data
+
+    # Device
+    @property
+    def device_ids(self):
+        """Returns external USB storage device ids."""
+        devices = []
+        for device in self.get_devices:
+            devices.append(device)
+        return devices
+
+    def get_device(self, device_id) -> SynoCoreExternalUSBDevice:
+        """Returns a specific external USB storage device."""
+        return self._data[device_id]
+
+
+class SynoCoreExternalUSBDevice:
+    """A representation of an external USB device."""
+
+    def __init__(self, data):
+        """Initialize a external USB device."""
+        self._data = data
+        for partition_data in data["partitions"]:
+            self._data[partition_data["name_id"]] = SynoUSBStoragePartition(
+                partition_data
+            )
+
+    @property
+    def device_id(self):
+        """Return id of the device."""
+        return self._data["dev_id"]
+
+    @property
+    def device_name(self):
+        """The title of the external USB storage device."""
+        return self._data["dev_title"]
+
+    @property
+    def device_type(self):
+        """The type of the external USB storage device."""
+        return self._data["dev_type"]
+
+    def device_size_total(self, human_readable=False):
+        """Total size of the external USB storage device."""
+        return_data = int(self._data["total_size_mb"])
+        return_data = SynoFormatHelper.megabytes_to_bytes(return_data)
```

Comment:
```suggestion
        return_data = megabytes_to_bytes(int(self._data["total_size_mb"]))
```
example for omit `SynoFormatHelper.`